src/_pytest/python_api.py

```diff
@@ -38,6 +39,36 @@ def _non_numeric_type_error(value, at: Optional[str]) -> TypeError:
     )
 
 
+def _generate_approx_error_message(
+    full_object: object,
+    message_data: typing.List[typing.List[str]],
```

Comment:
Please import directly from `typing` instead of importing it:

```suggestion
    message_data: List[List[str]],
```

Same for all the other type annotations.